src/app/features/tech-record/components/tech-record-summary-changes/tech-record-summary-changes.component.spec.ts

```diff
@@ -0,0 +1,220 @@
+
+let actions$: ReplaySubject<Action>;
+let store: MockStore;
+describe('TechRecordSummaryChangesComponent', () => {
+  let component: TechRecordSummaryChangesComponent;
+  let fixture: ComponentFixture<TechRecordSummaryChangesComponent>;
+
+  beforeEach(async () => {
+    actions$ = new ReplaySubject<Action>();
+    await TestBed.configureTestingModule({
+      declarations: [TechRecordSummaryChangesComponent],
+      imports: [DynamicFormsModule, HttpClientTestingModule, RouterTestingModule, SharedModule],
+      providers: [
+        provideMockActions(() => actions$),
+        provideMockStore({ initialState: initialAppState }),
+        {
+          provide: RouterService,
+          useValue: {
+            getRouteNestedParam$(param: string) {
+              if (param === 'systemNumber') return of('123456');
+              if (param === 'createdTimestamp') return of('123123123');
+              return of('');
+            },
+          },
+        },
+      ],
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(TechRecordSummaryChangesComponent);
+    store = TestBed.inject(MockStore);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  describe('navigateUponSuccess', () => {
+    it('should handle state management', () => {
+      jest.spyOn(component.actions$, 'pipe');
+      component.navigateUponSuccess();
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      expect(component.actions$.pipe).toHaveBeenCalled();
+    });
+  });
+
+  describe('ngOnInit', () => {
+    it('should call navigateOnSuccess and initSubscriptions', () => {
+      jest.spyOn(component, 'navigateUponSuccess');
+      jest.spyOn(component, 'initSubscriptions');
+    });
+  });
+
+  describe('initSubscriptions', () => {
+    beforeEach(() => {
+      jest.spyOn(store, 'select');
+      component.ngOnInit();
+    });
+    it('should grab techRecord from the store', () => {
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      expect(store.select).toHaveBeenCalledWith(techRecord);
+    });
+    it('should grab editingTechRecord from the store', () => {
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      expect(store.select).toHaveBeenCalledWith(editingTechRecord);
+    });
+    it('should grab selectTechRecordChanges from the store', () => {
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      expect(store.select).toHaveBeenCalledWith(selectTechRecordChanges);
+    });
+    it('should grab selectTechRecordDeletions from the store', () => {
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      expect(store.select).toHaveBeenCalledWith(selectTechRecordDeletions);
+    });
+  });
+
+  describe('ngOnDestroy', () => {
+    it('should call the correct functions for onDestroy', () => {
```

Comment:
Can I suggest a rename for this, if i. ran this test and it failed i wouldn't know why it failed, id probably ask, what are the correct functions ?